classes/class-wc-connect-api-client.php

```diff
@@ -530,6 +557,22 @@ protected function authorization_header() {
 			return $authorization;
 		}
 
+		protected function request_signature_wccom( $token_secret, $endpoint, $method ) {
+			$request_url = WC_Helper_API::url( $endpoint );
+
+			$data = array(
+				'host'        => parse_url( $request_url, PHP_URL_HOST ), // host URL.
+				'request_uri' => parse_url( $request_url, PHP_URL_PATH ), // endpoint URL.
+				'method'      => $method,
+			);
+
+			if ( in_array( $method, array( 'POST', 'PUT' ), true ) ) {
+				$data['body'] = $this->request_body();
```

Comment:
Is this still setting the body to the Connect server request body here? I was thinking we'll need to pass the WCCOM request body into this function, but wasn't sure if this was handling that or using the Connect server request body?

Reply:
I think what you're getting at here is that if we are only doing a GET request to /subscriptions/ on WCCOM it really shouldn't need to include $data['body'] since that won't exist for only getting subscriptions. I added this for a bit of future-proofing since we can safely assume we will be adding PUT/POST requests in the future to WCCOM. Premature? 🤔

Reply:
No, I think you're exactly right to anticipate future needs here. To give this a more concrete example, if for some reason we need to send an activation request to WCCOM, we'll need to pass along a body of: `{ 'product_key': 12345 }` like in here: https://github.com/woocommerce/woocommerce/blob/8db66f705f5906ff8e6f7f105c095fb71774dc40/includes/admin/helper/class-wc-helper.php#L1345-L1349

In this current implementation, I think the `$data['body']` would be set to our Connect server body, for example the settings payload, rather than just containing the product key to activate. The signature generation will only care about the product key body we'll be sending to WCCOM not the settings payload we're sending to Connect server.

That's why I was thinking we could pass in the body to the function for the most flexibility going forward and have it default to false or empty and skip setting `$data['body']` in those cases.

Reply:
How about this implementation?

https://github.com/Automattic/woocommerce-services/pull/2245/commits/40eb8390575bf1a21f6193c54eb7c531278ec871

I've tested with the site connected and disconnected from WC.com as well as the /subscriptions/ endpoint and other areas of WCS(label printing, getting services) and it all continues to work without requiring a WC.com connection.

Reply:
Looks good to me!